client/webserver/site/src/js/charts.ts

```diff
@@ -1278,33 +1272,35 @@ class Region {
     // with this transform in place using data coordinates, and remove the
     // transform before stroking. The dataCoords method of the supplied tool
     // provides this functionality.
-    const yRange = dataExtents.yRange
-    const xFactor = region.xRange / dataExtents.xRange
-    const yFactor = region.yRange / yRange
-    const xMin = dataExtents.x.min
-    const yMin = dataExtents.y.min
-    // These translation factors are complicated because the (0, 0) of the
-    // region is not necessarily the (0, 0) of the canvas.
-    const tx = (region.x.min + xMin) - xMin * xFactor
-    const ty = -region.y.min - (yRange - yMin) * yFactor
-    const setTransform = () => {
-      // Data coordinates are flipped about y. Flip the coordinates and
-      // translate top left corner to canvas (0, 0).
-      ctx.transform(1, 0, 0, -1, -xMin, yMin)
-      // Scale to data coordinates and shift into place for the region's offset
-      // on the canvas.
-      ctx.transform(xFactor, 0, 0, yFactor, tx, ty)
-    }
-    // dataCoords allows some drawing to be performed directly in data
-    // coordinates. Most actual drawing functions like ctx.stroke and
-    // ctx.fillRect should not be called from inside dataCoords, but
-    // ctx.moveTo and ctx.LineTo are fine.
-    tools.dataCoords = f => {
-      ctx.save()
-      setTransform()
-      f()
-      ctx.restore()
-    }
+
+    // TODO: Figure out why this doesn't work on WebView.
```

Comment:
What is the effect of commenting all this out?

Reply:
We can't plot directly in data coordinates any more. We have to use `tools.x` and `tools.y` instead.